### DISCUSSION TITLE
Pre-commit hook failing; is it fine to force-commit and fix later?

### DISCUSSION THREAD

**Discussion by lynnpepin** [st3v3nmw/obsidian-spaced-repetition#1170] [Q&A]
Hi, I'm working on a fork of OSR that adds a "reshuffle" button. I don't have any expectations that my fork will be merged into the plugin, but I'd like to keep my fork to a high enough standard that it could be merged :)

When I attempt a `git commit`, I get the following error:

```
[test]  FAIL  tests/unit/utils/dates.test.ts (5.571 s)
[test]   ● Format date › Different input overloads
[test]
[test]     expect(received).toBe(expected) // Object.is equality
[test]
[test]     Expected: "2023-01-01"
[test]     Received: "2022-12-31"
[test]
[test]        5 |         expect(formatDate(new Date(2023, 0, 1))).toBe("2023-01-01");
[test]        6 |         expect(formatDate(2023, 1, 1)).toBe("2023-01-01");
[test]     >  7 |         expect(formatDate(1672531200000)).toBe("2023-01-01");
[test]          |                                           ^
[test]        8 |     });
[test]        9 |
[test]       10 |     test("handles a leap year date", () => {
[test]
[test]       at Object.<anonymous> (tests/unit/utils/dates.test.ts:7:43)
```

I suspect this test is failing due to a package / version problem, or some machine-specific date issue... 

But before I try to debug and fix it, I'd like to first `git commit --no-verify`, and then fix this in a later commit. **Is this acceptable, or should all commits pass all tests?** This was not covered in [CONTRIBUTING.md](https://github.com/st3v3nmw/obsidian-spaced-repetition/blob/master/docs/docs/en/contributing.md), hence my question :)

---

**TLDR: When contributing, is it okay to `git commit -n` to bypass failing tests, then fix the problem in a later commit?**
